app/models/course/leaderboard_settings.rb

```diff
@@ -35,13 +35,48 @@ def display_user_count
     @settings.display_user_count || 30
   end
 
-  # Returns the number of users to be displayed on the leaderboard
+  # Set the number of users to be displayed on the leaderboard
   #
   # @param [Integer] count The number of users to be displayed
   def display_user_count=(count)
     @settings.display_user_count = count
   end
 
+  # Returns whether group leaderboard is enabled (disabled by default).
+  #
+  # @return [Boolean] option Setting on whether group leaderboard is enabled.
+  def display_group_leaderboard
+    @settings.display_group_leaderboard = false if @settings.display_group_leaderboard.nil?
```

Comment:
@weiqingtoh I would prefer to nest the group leaderboard settings under it's own key:
```
@group_leaderboard_settings = @settings.settings(:group_leaderboard)
@group_leaderboard_settings.display = true/false
@group_leaderboard_settings.title = 'Title'
```

what do you think ?